src/debug.c

```diff
@@ -1865,11 +1868,10 @@ __attribute__ ((noinline)) static void collect_stacktrace_data(void) {
 __attribute__ ((noinline))
 static void writeStacktraces(int fd, int uplevel) {
     /* get the list of all the process's threads that don't block or ignore the THREADS_SIGNAL */
-    pid_t pid = getpid();
     pid_t tids[TIDS_MAX_SIZE];
-    size_t len_tids = get_ready_to_signal_threads_tids(pid, THREADS_SIGNAL, tids);
+    size_t len_tids = get_ready_to_signal_threads_tids(THREADS_SIGNAL, tids);
     if (!len_tids) {
-        serverLogFromHandler(LL_WARNING, "writeStacktraces(): Failed to get the process's threads.");
+        serverLogRawFromHandler(LL_WARNING, "writeStacktraces(): Failed to get the process's threads.");
```

Comment:
If `get_ready_to_signal_threads_tids()` returns zero, rest of this function is going to work just fine right? 
e.g. we try to open() directory inside `get_ready_to_signal_threads_tids()`. I remember something like opening directory not supported on some (weird) systems and this function might return zero.